### PR TITLE
Support commercial cards

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #112 #113 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

The scope of this update adds the new feature to support corp credit cards with less than 16 digits and a new `27` numeration for the Alaska card icon. 

This PR also refactors the use of the regex per credit card supported. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/base-input.js [hunk 1]
```diff
@@ -596,37 +602,50 @@ export default class BaseInput extends LitElement {
...
         customValidationMessage: defaultCustomValidationMessage,
         cardIcon: 'cc-discover'
       },
+      {
+        name: 'Alaska Commercial',
+        regex: /^(?<num>27)\d{0}/u,
+        formatMinLength: 8,
+        customValidationMessage: defaultCustomValidationMessage,
+        cardIcon: 'cc-alaska'
+      },
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply src/base-input.js:645]: I figured it out. The problem is a missing line. The `Commercial` card type is missing the cardIcon attribute to set the default card.

File: src/base-input.js [hunk 2]
```diff
@@ -596,37 +602,50 @@ export default class BaseInput extends LitElement {
 
     // eslint-disable-next-line sort-vars, one-var
     const creditCardTypes = [
+      {
+        name: 'Commercial',
+        regex: /^(?<num>2)\d{0}/u,
+        formatMinLength: 8,
+        customValidationMessage: defaultCustomValidationMessage
+      },
```
Comment (jason-capsule42) [inline src/base-input.js:610]: See comment below for why this is necessary

```suggestion
        customValidationMessage: defaultCustomValidationMessage,
        cardIcon: 'credit-card'
      },
```

File: test/auro-input.test.js [hunk 3]
```diff
@@ -290,14 +290,23 @@ describe('auro-input', () => {
...
     });
+
+    it('Alaska Corp Cards', async () => {
+      const el = await fixture(html`
+        <auro-input id="format-ccWithIcon" type="credit-card" icon label="Credit Card Number with Icon" required></auro-input>
+      `);
+
+      await setInputValue(el, '27');
+      expect(el.shadowRoot.querySelector('.accentIcon')).to.have.attribute('name', 'cc-alaska');
+    });
```
Comment (jason-capsule42) [inline test/auro-input.test.js:309]: We should have a similar test for just the corporate card match value of `2`.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: test/auro-input.test.js (hunk 3 above)
Comment (blackfalcon) [reply test/auro-input.test.js:318]: Test added

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**